frontend/public/components/monitoring.tsx

```diff
@@ -1840,22 +1842,28 @@ type PrometheusAlert = {
   labels: PrometheusLabels & {
     alertname: string;
   };
-  state: AlertStates;
-  value?: number;
+  state: AlertStates | string;
+  value?: number | string;
+  partialResponseStrategy?: string;
 };
 
 export type Alert = PrometheusAlert & {
   rule: Rule;
   silencedBy?: Silence[];
 };
 
-type PrometheusRule = {
-  alerts: PrometheusAlert[];
-  annotations: PrometheusLabels;
-  duration: number;
+export type PrometheusRule = {
+  alerts?: PrometheusAlert[];
```

Comment:
`PrometheusRule` represents the format of an alerting rule as returned by the Prometheus API (https://prometheus.io/docs/prometheus/latest/querying/api/#rules). So any updates here should only be to better match the Prometheus API.

Reply:
Ok, I think it would be better to define another type for ```ThanosRule```.

Reply:
👍